Jenkinsfile

```diff
@@ -171,11 +188,31 @@ pipeline {
         }
         stage('build') {
             steps {
+                withCredentials([usernamePassword(
+                    credentialsId: params.DOCKER_CREDENTIALS,
+                    usernameVariable: 'DOCKER_HUB_USERNAME',
+                    passwordVariable: 'DOCKER_HUB_PASSWORD',
+                )]) {
+                    sh '''
+                        if [ -n "${FISSILE_DOCKER_REGISTRY}" ]; then
+                            docker login -u "${DOCKER_HUB_USERNAME}" -p "${DOCKER_HUB_PASSWORD}" "${FISSILE_DOCKER_REGISTRY}"
+                        fi
+                    '''
+                }
                 sh '''
                     set -e +x
                     source ${PWD}/.envrc
                     set -x
                     unset HCF_PACKAGE_COMPILATION_CACHE
+
+                    # Make sure to use the .envrc variables if the override variables are empty
+                    if [ -z "${FISSILE_STEMCELL}" ]; then
+                        unset FISSILE_STEMCELL
```

Comment:
But you're doing all this _after_ sourcing `.envrc` ~6 lines above.

Also note that `.envrc` mostly uses `${FOO:-default}` anyway, which uses the default value if there's no override.  Meaning you don't actually need this block at all.

Reply:
Nice catch. 

And we thought `${FOO:-default}` would only fall back to the default if `FOO` isn't set at all, but not if it's empty. That's good to know.